pkg/build/builder/docker.go

```diff
@@ -403,3 +440,25 @@ func insertEnvAfterFrom(node *parser.Node, env []kapi.EnvVar) error {
 
 	return nil
 }
+
+// getDockerfilefrom returns all the value of "FROM" instruction inside the dockerfile
+func getDockerfileFrom(dockerfilePath string) []string {
+	var froms []string
+	if "" == dockerfilePath {
+		return froms
+	}
+	node, err := parseDockerfile(dockerfilePath)
+	if err != nil {
+		return froms
+	}
+	for i := len(node.Children) - 1; i >= 0; i-- {
+		child := node.Children[i]
+		if child != nil && child.Value == dockercmd.From {
+			from := child.Next.Value
+			if from != "" {
+				froms = append([]string{from}, froms...)
```

Comment:
I would recommend doing a normal loop and then reversing it at the end, vs this (which is more complex to read)